--- a/src/basic/missing_prctl.h
+++ b/src/basic/missing_prctl.h
@@ -1,6 +1,22 @@
 /* SPDX-License-Identifier: LGPL-2.1-or-later */
 #pragma once

+#include <linux/prctl.h>
+
+#if defined(__linux__) && defined(PR_CAP_AMBIENT) && defined(__PRCTL_MM_MAP_DEFINED)
+#undef __PRCTL_MM_MAP_DEFINED
+#endif
+
+#ifndef __PRCTL_MM_MAP_DEFINED
+struct prctl_mm_map {
+    unsigned long start_code;
+    unsigned long end_code;
+    unsigned long start_data;
+    unsigned long end_data;
+    unsigned long start_brk;
+    unsigned long brk;
+    unsigned long start_stack;
+    unsigned long arg_start;
+    unsigned long arg_end;
+    unsigned long env_start;
+    unsigned long env_end;
+};
+#define __PRCTL_MM_MAP_DEFINED
+#endif
 #include "macro.h"
