--- a/src/basic/missing_prctl.h	2024-10-08 23:42:30.000000000 +0800
+++ a/src/basic/missing_prctl.h	2024-12-19 02:42:30.404342887 +0800
@@ -2,6 +2,25 @@
 #pragma once
 
 #include <linux/prctl.h>
+
+#if defined(__linux__) && !defined(HAVE_PRCTL_MM_MAP) && !defined(__PRCTL_MM_MAP_DEFINED)
+#define HAVE_PRCTL_MM_MAP
+
+struct prctl_mm_map {
+    unsigned long start_code;
+    unsigned long end_code;
+    unsigned long start_data;
+    unsigned long end_data;
+    unsigned long start_brk;
+    unsigned long brk;
+    unsigned long start_stack;
+    unsigned long arg_start;
+    unsigned long arg_end;
+    unsigned long env_start;
+    unsigned long env_end;
+};
+#define __PRCTL_MM_MAP_DEFINED
+#endif
 
 #include "macro.h"
 
